.github/workflows/build_servers_pr.yml

```diff
@@ -0,0 +1,95 @@
+name: Build Servers from a Pull Request
+# build_servers_pr.yml
+
+on:
+  pull_request:
+    paths:
+      # This workflow should run when a file in the src/ directory has been modified.
+      - 'src/**'
+      # If changes have only been made in one of the following directories, the workflow will not run.
+      # Additional workflow Actions can be prepared for NPC script tests.
+      - '!conf/**'
+      - '!db/**'
+      - '!doc/**'
+      - '!npc/**'
+
+jobs:
+  build:
+    # Github Actions checks for 'ci skip', 'skip ci', but not a hyphenated version.
+    # It's a catch-all incase a Pull Request has been opened and someone is on auto-pilot.
+    if: "!contains(github.event.head_commit.message, 'ci-skip')"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+          # The ubuntu-latest label currently points to ubuntu-18.04.
+          # Available: ubuntu-20.04, ubuntu-18.04	
+          os: [ubuntu-latest]
+          # There is no reason to check build success for packet-versions older than four years.
+          packetver: ['20211103', '20200902', '20200401', '20180620']
+          # Older versions of GCC are not available via unaltered aptitude repo lists.
+          gcc: ['7', '8', '9', '10']
+          # We run build checks for both Renewal and PRE-Renewal
+          prere: ['PRE', 'RE']
```

Comment:
```suggestion
          mode: ['PRE', 'RE']
```

Reply:
I'm not sure why I didn't just use mode in the first place - I guess because of the --enable-prere I wanted to keep it relative..